src/cascadia/TerminalSettingsModel/CascadiaSettings.h

```diff
@@ -20,163 +20,135 @@ Author(s):
 #include "CascadiaSettings.g.h"
 
 #include "GlobalAppSettings.h"
-#include "TerminalWarnings.h"
-#include "IDynamicProfileGenerator.h"
 
 #include "Profile.h"
 #include "ColorScheme.h"
 
-// fwdecl unittest classes
-namespace SettingsModelLocalTests
-{
-    class SerializationTests;
-    class DeserializationTests;
-    class ProfileTests;
-    class ColorSchemeTests;
-    class KeyBindingsTests;
-};
-namespace TerminalAppUnitTests
+namespace winrt::Microsoft::Terminal::Settings::Model::implementation
 {
-    class DynamicProfileTests;
-    class JsonTests;
-};
+    winrt::com_ptr<Profile> ReproduceProfile(const winrt::com_ptr<Profile>& parent);
 
-namespace Microsoft::Terminal::Settings::Model
-{
-    class SettingsTypedDeserializationException;
-};
+    class SettingsTypedDeserializationException final : public std::runtime_error
+    {
+    public:
+        SettingsTypedDeserializationException(const char* message) noexcept :
+            std::runtime_error(message) {}
+    };
 
-class Microsoft::Terminal::Settings::Model::SettingsTypedDeserializationException final : public std::runtime_error
-{
-public:
-    SettingsTypedDeserializationException(const std::string_view description) :
-        runtime_error(description.data()) {}
-};
+    struct ParsedSettings
+    {
+        winrt::com_ptr<implementation::GlobalAppSettings> globals;
+        winrt::com_ptr<implementation::Profile> baseLayerProfile;
+        std::vector<winrt::com_ptr<implementation::Profile>> profiles;
+        std::unordered_map<winrt::guid, winrt::com_ptr<implementation::Profile>> profilesByGuid;
+    };
+
+    struct SettingsLoader
+    {
+        static SettingsLoader Default(const std::string_view& userJSON, const std::string_view& inboxJSON);
+        SettingsLoader(const std::string_view& userJSON, const std::string_view& inboxJSON);
+
+        void GenerateProfiles();
+        void FillBlanksInDefaultsJson();
+        void MergeInboxIntoUserProfiles();
+        void MergeFragmentsIntoUserProfiles();
+        void DisableDeletedProfiles();
+        void FinalizeLayering();
+
+        ParsedSettings inboxSettings;
+        ParsedSettings userSettings;
+        bool duplicateProfile = false;

```

Comment:
This class is an internal implementation detail of `CascadiaSettings`. Since there's no "bad actor" messing with the state of this class, a getter would then only be unnecessarily "verbose" in my personal opinion.